source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,228 @@
+.. _csharp-tls:
+
+==========================
+Enable TLS on a Connection
+==========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to connect to MongoDB instances with the 
+:wikipedia:`TLS/SSL <Transport_Layer_Security>`
+security protocol using the underlying TLS/SSL support in the {+framework+}. To
+configure your connection to use TLS/SSL, enable the TLS/SSL settings in either
+the :ref:`connection string <csharp-connection-uri>` or 
+:ref:`MongoClientSettings <csharp-mongo-client-settings>`.
+
+.. _csharp-tls-enable:
+
+Enable TLS
+----------
+
+You can enable TLS for the connection to your MongoDB instance
+in two different ways: using a property on a ``MongoClientSettings`` object or
+through a parameter in your connection string.
+
+.. tabs::
+
+   .. tab:: MongoClientSettings
+      :tabid: mongoclientsettings
+
+      To enable TLS with a ``MongoClientSettings`` object, set the ``UseTls`` property
+      to ``true``:
+
+      .. code-block:: csharp
+
+         var settings = new MongoClientSettings() { UseTls = true };
+         var client = new MongoClient(settings);
+
+   .. tab:: Connection String
+      :tabid: connectionstring
+
+      To enable TLS with a connection string, assign the 
+      parameter ``tls`` a value of ``true`` in the connection string passed to the
+      ``MongoClient`` constructor:
+
+      .. code-block:: csharp
+
+         var mongoClient = new MongoClient("mongodb://<username>:<password>@<hostname>:<port>?tls=true");
+
+Configure a Client Certificate
+------------------------------
+
+You can configure your X.509 certificate using ``MongoClientSettings``. The following
+code sample creates a new X.509 certificate object using the certificate file named 
+``client.p12``, which is protected by the password ``mySuperSecretPassword``. The code
+then adds this certificate to the ``SslSettings.ClientCertificates`` array in 
+``MongoClientSettings``.
+
+.. code-block:: csharp
+
+   var cert = new X509Certificate2("client.p12", "mySuperSecretPassword");
```

Comment:
In [another PR](https://github.com/mongodb/docs-csharp/pull/57#discussion_r1045291627), Dima suggested using .p12 certs instead of .pfx here. Can you confirm that my updates are correct?

Reply:
@DmitryLukyanov please :)

Reply:
as wiki [says](https://en.wikipedia.org/wiki/PKCS_12) `.p12` is the successor to `pfx` format. Nowadays these formats are interchangeable. As far as I know microsoft for historical reasons still prefers pfx extension, but I would say better to mention the most modern format :) We also use `p12` format in our tests